src/components/HourlyCityWeather/HourlyCityWeather.js

```diff
@@ -0,0 +1,168 @@
+import { useEffect, useState } from "react";
+import { Col, Row } from "react-bootstrap";
+import Carousel from 'react-bootstrap/Carousel';
+import './HourlyCityWeather.css';
+
+const HourlyCityWeather = ({city}) => {
+    const [error, setError] = useState(null);
+    const [isLoaded, setIsLoaded] = useState(false);
+    const [timeStamps, setTimeStamps] = useState(0);
+
+
+    useEffect(() => {
+        fetch(
+            "https://api.openweathermap.org/data/2.5/forecast?q=" +
+            city +
+            "&units=metric" +
+            "&appid=" +
+            process.env.REACT_APP_APIKEY
+        )
+            .then((res) => res.json())
+            .then(
+                (result) => {
+                    if (result["cod"] !== "200") {
+                        setIsLoaded(true);
+                        setError(result);
+                    } else {
+                        setIsLoaded(true);
+                        setError();
+                        organizeForecast(result);
+                    }
+                },
+                (error) => {
+                    setError(error);
+                }
+            )
+            .catch((err) => {
+                setError(err);
+            });
+
+    }, [city], []);
+
+    function organizeForecast(result) {
+        const hourlyForecasts = [];
+        var dailyData = [];
+        for (let i = 0; i < result.list.length; i++) {
+            var dateAndTime = result.list[i].dt_txt.split(" ");
+            var time = dateAndTime[1].split(":");
+            time = time[0] + ":" + time[1];
+            dailyData = {
+                "day": getWeekday(result.list[i].dt_txt),
+                "time": time,
+                "temperature": parseInt(result.list[i].main.temp, 10),
+                "icon": result.list[i].weather[0].main
+            };
+            hourlyForecasts.push(dailyData);
+        }
+        var dailyForecasts = [];
+        for (var j = 0; j < hourlyForecasts.length; j += 8) {
+            dailyForecasts.push(hourlyForecasts.slice(j, j + 8));
+        }
```

Comment:
You can use a similar solution as above here to build the `dailyForecasts` array. It's also a good idea to use `const` variables wherever possible, and when you need to mutate the variable usually `let` it better than `var`. To learn more check [this article](https://www.freecodecamp.org/news/var-let-and-const-whats-the-difference/) out.